sdk-core/src/main/java/care/data4life/sdk/network/ApiService.kt

```diff
@@ -0,0 +1,413 @@
+/*
+ * Copyright (c) 2021 D4L data4life gGmbH / All rights reserved.
+ *
+ * D4L owns all legal rights, title and interest in and to the Software Development Kit ("SDK"),
+ * including any intellectual property rights that subsist in the SDK.
+ *
+ * The SDK and its documentation may be accessed and used for viewing/review purposes only.
+ * Any usage of the SDK for other purposes, including usage for the development of
+ * applications/third-party applications shall require the conclusion of a license agreement
+ * between you and D4L.
+ *
+ * If you are interested in licensing the SDK for your own applications/third-party
+ * applications and/or if you’d like to contribute to the development of the SDK, please
+ * contact D4L by email to help@data4life.care.
+ */
+package care.data4life.sdk.network
+
+import care.data4life.auth.AuthorizationContract
+import care.data4life.sdk.NetworkConnectivityService
+import care.data4life.sdk.lang.D4LException
+import care.data4life.sdk.lang.D4LRuntimeException
+import care.data4life.sdk.network.model.CommonKeyResponse
+import care.data4life.sdk.network.model.DocumentUploadResponse
+import care.data4life.sdk.network.model.EncryptedRecord
+import care.data4life.sdk.network.model.NetworkModelContract
+import care.data4life.sdk.network.model.UserInfo
+import care.data4life.sdk.network.model.VersionList
+import care.data4life.sdk.network.typeadapter.EncryptedKeyTypeAdapter
+import care.data4life.sdk.util.Base64.encodeToString
+import com.squareup.moshi.Moshi
+import io.reactivex.Completable
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import okhttp3.CertificatePinner
+import okhttp3.Interceptor
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.Response
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+import java.io.IOException
+import java.net.SocketTimeoutException
+import java.util.HashMap
+import java.util.concurrent.TimeUnit
+
+// TODO Kotlin and internal
+class ApiService constructor(
+    private val authService: AuthorizationContract.Service,
+    private val environment: NetworkingContract.Environment,
+    private val clientID: String,
+    private val clientSecret: String,
+    private val platform: String?,
+    private val connectivityService: NetworkConnectivityService,
+    private val clientName: String?,
+    staticAccessToken: ByteArray?,
+    private val debug: Boolean
+) : NetworkingContract.Service {
+    private var service: IHCService? = null
```

Comment:
why is this nullable? couldn't it be initialized in the init() block?

Reply:
You are totally right about this...but I like not put effort into it, since this class had been already rewritten later in the current chain (I will tag the PR once it is uploaded). This step was only necessary to enable proper testing.

Reply:
Please see #139.